### PR TITLE
Rsocket impl

### LINKED ISSUE #41: Rsocket auth

### PR DESCRIPTION
**Pull request by l3r8yJ**
closes #41 

test implementation

@h1alexbel wdyt

---

## PR-Codex overview
This PR adds support for a hub, updates method names and adds license headers to package-info files.

### Detailed summary
- Added a `hub` section to `application.yaml` for hub configuration
- Renamed methods `messageAsBotCommand` to `asBotCommand` in `Conversation`, `Start`, and `Token` classes
- Added MIT license headers to `package-info.java` files in `request`, `response`, and `extension` packages

> The following files were skipped due to too many changes: `src/main/java/io/blamer/bot/reponse/RegistryResponse.java`, `src/test/java/io/blamer/bot/conversation/routes/RegistryTest.java`, `src/main/java/io/blamer/bot/extension/ExtRSocket.java`, `src/main/java/io/blamer/bot/conversation/routes/Registry.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/main/java/io/blamer/bot/configuration/RSocketConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import java.time.Duration;
+
+/**
+ * RSocket config.
+ */
+@Slf4j
+@Data
+@Configuration
+@ConfigurationProperties("hub")
+public class RSocketConfiguration {
```
Comment (h1alexbel) [inline src/main/java/io/blamer/bot/configuration/RSocketConfiguration.java:45]: @l3r8yJ I think `ExtRs` or `ExtRSocket` is a better naming for this object, like [this](https://github.com/Blamer-io/bot/blob/master/src/main/java/io/blamer/bot/agents/tg/ExtTg.java)

File: src/main/java/io/blamer/bot/conversation/routes/Registry.java [hunk 2]
```diff
@@ -56,67 +61,31 @@ public class Registry implements Conversation {
...
   private String description;
 
-  /*
-   * @todo #19 After hub service will be created, redirect the token.
-   *   Send auth message to hub, to process the data.
-   * */
+  /**
+   * The requester.
+   */
+  private final RSocketRequester requester;
```
Comment (h1alexbel) [inline src/main/java/io/blamer/bot/conversation/routes/Registry.java:67]: @l3r8yJ I suggest to name this var as `rs`
WDYT?

File: src/main/java/io/blamer/bot/conversation/routes/Registry.java [hunk 3]
```diff
@@ -56,67 +61,31 @@ public class Registry implements Conversation {
...
+        .replace("/registry", " ")
+        .strip();
+      return this.requester
+        .route("hub.auth")
+        .data(new RegistryRequest(token, chat))
+        .retrieveMono(RegistryResponse.class)
+        .map(auth -> new SendMessage(auth.getChat(), auth.getText()))
+        .block();
+    } catch (final RuntimeException ex) {
+      return new SendMessage(chat, "Internal error...");
```
Comment (h1alexbel) [inline src/main/java/io/blamer/bot/conversation/routes/Registry.java:83]: @l3r8yJ Also, maybe we should give a customer a chance to see full stacktrace?

WDYT

File: src/main/java/io/blamer/bot/request/RegistryRequest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import lombok.NoArgsConstructor;
+
+/**
+ * Request for auth.
+ */
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class RegistryRequest {
```
Comment (h1alexbel) [inline src/main/java/io/blamer/bot/request/RegistryRequest.java:39]: @l3r8yJ maybe `RqRegistry`?

WDYT?

File: src/test/resources/application.yaml [hunk 5]
```diff
@@ -1,3 +1,6 @@
 bot:
   token: ${BOT_TOKEN}
   name: ${BOT_NAME}
+hub:
+  host: localhost
```
Comment (h1alexbel) [inline src/test/resources/application.yaml:5]: @l3r8yJ should be ${HOST}, I beleive

File: src/test/resources/application.yaml [hunk 6]
```diff
@@ -1,3 +1,6 @@
...
+  port: 7070
```
Comment (h1alexbel) [inline src/test/resources/application.yaml:6]: @l3r8yJ port shouldn't be hardcoded

File: src/main/java/io/blamer/bot/conversation/routes/Registry.java (hunk 2 above)
Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:67]: @h1alexbel 2 chars will be violation for checkstyle

File: src/main/java/io/blamer/bot/conversation/routes/Registry.java (hunk 3 above)
Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:83]: @h1alexbel until we in dev mode, I think we can throw an exception message

File: src/main/java/io/blamer/bot/request/RegistryRequest.java (hunk 4 above)
Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/request/RegistryRequest.java:39]: @h1alexbel We have no interfaces to add these prefixes `Rq`/`Rs`. I think prefixes should be like markers

File: src/main/java/io/blamer/bot/conversation/routes/Registry.java (hunk 3 above)
Comment (h1alexbel) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:83]: @l3r8yJ in dev mode it also will help

File: src/main/java/io/blamer/bot/request/RegistryRequest.java (hunk 4 above)
Comment (h1alexbel) [reply src/main/java/io/blamer/bot/request/RegistryRequest.java:39]: @l3r8yJ maybe we should implement an interface `Registry`? > RqRegistry will be concrete class

**Comment by l3r8yJ** [comment]
@h1alexbel take a look, please

File: src/main/java/io/blamer/bot/conversation/routes/Registry.java (hunk 2 above)
Comment (h1alexbel) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:67]: @l3r8yJ sure, you are right

File: src/main/java/io/blamer/bot/request/RegistryRequest.java (hunk 4 above)
Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/request/RegistryRequest.java:39]: @h1alexbel i think better would be implement `Request`/`Response` for `Rq`/`Rs`, wdyt?

Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/request/RegistryRequest.java:39]: @h1alexbel but I suggest this refactoring when we need additional responses/requests

Comment (h1alexbel) [reply src/main/java/io/blamer/bot/request/RegistryRequest.java:39]: @l3r8yJ agree with you

File: src/main/java/io/blamer/bot/conversation/routes/Registry.java [hunk 7]
```diff
@@ -56,67 +61,34 @@ public class Registry implements Conversation {
...
+      return this.requester
+        .route("hub.auth")
+        .data(new RegistryRequest(token, chat))
+        .retrieveMono(RegistryResponse.class)
+        .map(auth -> new SendMessage(auth.getChat(), auth.getText()))
+        .block();
+    } catch (final RuntimeException ex) {
+      return new SendMessage(
+        chat,
+        "Internal error: %s".formatted(ex.getMessage())
```
Comment (h1alexbel) [inline src/main/java/io/blamer/bot/conversation/routes/Registry.java:85]: @l3r8yJ  how about full error stackstrace  in the format `like this`

Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:85]: @h1alexbel didn't provide source

Comment (h1alexbel) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:85]: @l3r8yJ, I mean bash `` formatting

Comment (h1alexbel) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:85]: @l3r8yJ, I mean bash `` formatting

Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:85]: @h1alexbel just put `ex.getMessage()` or i miss something?

Comment (h1alexbel) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:85]: Yes, and apply bash formatting.
For Telegram it's ``` ${text} ```

Comment (h1alexbel) [reply src/main/java/io/blamer/bot/conversation/routes/Registry.java:85]: Yes, and apply bash formatting.
For Telegram it's ``` ${text} ```

**Comment by l3r8yJ** [comment]
@h1alexbel now should have formatting like bash

**Review by h1alexbel — APPROVED** [review]
lgtm

**Comment by l3r8yJ** [comment]
@rultor merge

**Comment by l3r8yJ** [comment]
@rultor merge

**Comment by l3r8yJ** [comment]
@h1alexbel i guess here we can only disable`codecov` temporary or manually merge, wdyt

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**